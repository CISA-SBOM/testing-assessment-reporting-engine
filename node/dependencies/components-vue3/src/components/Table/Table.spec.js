import { describe, expect, it } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import Component from "./Table.vue";

describe("Table.vue", () => {
  it("matches snapshot with no props assigned", () => {
    const props = {};
    const wrapper = shallowMount(Component, { props });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with items and action col props assigned", () => {
    const props = {
      items: [
        {
          id: 1,
          name: "A title",
          createdDate: new Date("2000-01-01"),
          lastUpdatedDate: new Date("2014-11-12"),
        },
        {
          id: 2,
          name: "B title",
          createdDate: new Date("2013-02-01"),
          lastUpdatedDate: new Date("2013-10-10"),
        },
      ],
      sortBy: "lastUpdatedDate",
      fields: [
        { key: "name", label: "Title", sortable: true },
        { key: "createdDate", label: "Created", sortable: true, format: (date) => date.toLocaleDateString() },
        {
          key: "lastUpdatedDate",
          label: "Last modified",
          sortable: true, format: (date) => date.toLocaleDateString()
        },
      ],
    };
    const wrapper = shallowMount(Component, { props });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with entries but no action col props assigned", () => {
    const props = {
      items: [
        {
          id: 1,
          name: "A title",
          createdDate: new Date("2000-01-01"),
          lastUpdatedDate: new Date("2014-11-12"),
        },
        {
          id: 2,
          name: "B title",
          createdDate: new Date("2013-02-01"),
          lastUpdatedDate: new Date("2013-10-10"),
        },
      ],
      fields: [
        { key: "name", label: "Title", sortable: true },
        { key: "createdDate", label: "Created", sortable: true, format: (date) => date.toLocaleDateString() },
        {
          key: "lastUpdatedDate",
          label: "Last modified",
          sortable: true, format: (date) => date.toLocaleDateString()
        },
        { key: "actions", label: "Actions" }
      ]
    };
    const slots = {
      'cell(actions)': `
        <template #cell(actions)="{item}">
          <button
            class="btn btn-link text-red-300 btn-sm"
            @click="remove(entry)"
          >
            <i class="fas fa-trash"></i>
            Remove entry
          </button>
        </template>
      `,
    };
    const wrapper = shallowMount(Component, { props, slots });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
