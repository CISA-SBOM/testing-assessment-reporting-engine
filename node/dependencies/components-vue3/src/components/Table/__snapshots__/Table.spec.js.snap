// Vitest Snapshot v1

exports[`Table.vue > matches snapshot with entries but no action col props assigned 1`] = `
"<table data-id=\\"sds-table\\">
  <!---->
  <colgroup>
    <col class=\\"\\">
    <col class=\\"\\">
    <col class=\\"\\">
    <col class=\\"\\">
  </colgroup>
  <thead>
    <tr>
      <th class=\\"cursor-pointer space-x-1 select-none group\\">Title <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-50\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\">
          <path d=\\"M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z\\" fill=\\"currentColor\\"></path>
          <!---->
          <!---->
        </svg></th>
      <th class=\\"cursor-pointer space-x-1 select-none group\\">Created <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-50\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\">
          <path d=\\"M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z\\" fill=\\"currentColor\\"></path>
          <!---->
          <!---->
        </svg></th>
      <th class=\\"cursor-pointer space-x-1 select-none group\\">Last modified <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-50\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\">
          <path d=\\"M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z\\" fill=\\"currentColor\\"></path>
          <!---->
          <!---->
        </svg></th>
      <th class=\\"space-x-1 select-none group\\">Actions
        <!---->
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>A title</td>
      <td>12/31/1999</td>
      <td>11/11/2014</td>
      <td><button class=\\"btn btn-link text-red-300 btn-sm\\"><i class=\\"fas fa-trash\\"></i> Remove entry </button></td>
    </tr>
    <tr>
      <td>B title</td>
      <td>1/31/2013</td>
      <td>10/9/2013</td>
      <td><button class=\\"btn btn-link text-red-300 btn-sm\\"><i class=\\"fas fa-trash\\"></i> Remove entry </button></td>
    </tr>
  </tbody>
</table>"
`;

exports[`Table.vue > matches snapshot with items and action col props assigned 1`] = `
"<table data-id=\\"sds-table\\">
  <!---->
  <colgroup>
    <col class=\\"\\">
    <col class=\\"\\">
    <col class=\\"null\\">
  </colgroup>
  <thead>
    <tr>
      <th class=\\"cursor-pointer space-x-1 select-none group\\">Title <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-50\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\">
          <path d=\\"M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z\\" fill=\\"currentColor\\"></path>
          <!---->
          <!---->
        </svg></th>
      <th class=\\"cursor-pointer space-x-1 select-none group\\">Created <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-50\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\">
          <path d=\\"M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z\\" fill=\\"currentColor\\"></path>
          <!---->
          <!---->
        </svg></th>
      <th class=\\"null cursor-pointer space-x-1 select-none group\\">Last modified <svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" aria-hidden=\\"true\\" role=\\"img\\" class=\\"inline-block w-4 h-4 group-hover:opacity-100 opacity-100\\" preserveAspectRatio=\\"xMidYMid meet\\" viewBox=\\"0 0 320 512\\">
          <!---->
          <path d=\\"M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z\\" fill=\\"currentColor\\"></path>
          <!---->
        </svg></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>B title</td>
      <td>1/31/2013</td>
      <td>10/9/2013</td>
    </tr>
    <tr>
      <td>A title</td>
      <td>12/31/1999</td>
      <td>11/11/2014</td>
    </tr>
  </tbody>
</table>"
`;

exports[`Table.vue > matches snapshot with no props assigned 1`] = `
"<table data-id=\\"sds-table\\">
  <!---->
  <colgroup></colgroup>
  <thead>
    <tr></tr>
  </thead>
  <tbody></tbody>
</table>"
`;
